src/runtime/code.cpp

```diff
@@ -200,11 +200,17 @@ extern "C" BORROWED(PyObject*) PyCode_GetFilename(PyCodeObject* op) noexcept {
     RELEASE_ASSERT(PyCode_Check((Box*)op), "");
     return BoxedCode::filename((Box*)op, NULL);
 }
+
 extern "C" BORROWED(PyObject*) PyCode_GetName(PyCodeObject* op) noexcept {
     RELEASE_ASSERT(PyCode_Check((Box*)op), "");
     return BoxedCode::name((Box*)op, NULL);
 }
 
+extern "C" int PyCode_GetNumFree(PyCodeObject* _code) noexcept {
+    BoxedCode* code = (BoxedCode*)_code;
+    return code->f->source->getScopeInfo()->takesClosure();
```

Comment:
It looks like this doesn't actually return the number of free variables, but rather whether there are any.  Can we change the function to `bool PyCode_HasFreeVars`?